Make GridSplitter scaling aware.

grokys opened the pull request:
## What does the pull request do?

Fixes #8314: when comparing sizes in `GridSplitter.MoveSplitter`, account for `UseLayoutRounding` and DPI scaling by using an epsilon that is the size of a device pixel.

Unfortunately I've been unable to create a unit test that demonstrates the problem based on the repro, but I've checked manually and this seems to fix the problem.

## Fixed issues

Fixes #8314

grokys committed:
Make GridSplitter scaling aware.

When comparing sizes in `GridSplitter.MoveSplitter`, account for `UseLayoutRounding` and DPI scaling by using an epsilon that is the size of a device pixel.

avaloniaui-team commented:
You can test this PR using the following package version. `0.10.999-cibuild0021132-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

grokys committed:
Read scaling when starting a drag operation.

Previous implementation didn't update the scaling when moving between monitors.

avaloniaui-team commented:
You can test this PR using the following package version. `0.10.999-cibuild0021136-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

grokys committed:
Merge branch 'master' into fixes/8314-gridsplitter-layoutrounding
